cluster/terraform_aks_cluster/analytics.tf

```diff
@@ -39,3 +39,50 @@ resource "azurerm_monitor_diagnostic_setting" "aks_system_logs" {
     enabled  = false
   }
 }
+
+# Using a data collection rule to change the Container Insights data collection settings
+# see https://learn.microsoft.com/en-us/azure/azure-monitor/containers/container-insights-data-collection-dcr?tabs=cli
+
+resource "azurerm_monitor_data_collection_rule" "aks" {
+  name                = "${var.resource_prefix}-tsc-${var.environment}-dcr"
+  resource_group_name = data.azurerm_resource_group.cluster.name
+  location            = data.azurerm_resource_group.cluster.location
+
+  lifecycle { ignore_changes = [tags] }
+
+  destinations {
+    log_analytics {
+      name                  = "${var.resource_prefix}-tsc-${var.environment}-log-analytics-destination"
+      workspace_resource_id = azurerm_log_analytics_workspace.aks_system_logs.id
+    }
+  }
+
+  data_flow {
+    # streams      = ["Microsoft-ContainerInsights-Group-Default"]
+    streams      = ["Microsoft-ContainerLog", "Microsoft-ContainerLogV2"]
+    destinations = ["${var.resource_prefix}-tsc-${var.environment}-log-analytics-destination"]
+  }
+
+  data_sources {
+    extension {
+      name = "ContainerInsightsExtension"
+      # see https://learn.microsoft.com/en-us/azure/azure-monitor/containers/container-insights-transformations#data-sources
+      # streams        = ["Microsoft-ContainerInsights-Group-Default"]
```

Comment:
As discussed, let's keep insights for now and review later

Reply:
Changed steams to Microsoft-ContainerInsights-Group-Default